### PR TITLE
Add ray.util package and move libraries from experimental

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

This moves a number of libraries from experimental to util, updating the documentation.

### CODE REVIEW

**Comment by edoakes** [comment]
What are the -300 LOC? Removing some documentation?

**Comment by ericl** [comment]
Removing deprecated things.

**Comment by ericl** [comment]
Specifically, signals (documentation only) and the GCS flushing code was completely removed rather than moved.

**Review by edoakes — APPROVED** [review]
LGTM, lots of conflicts with master though

**Comment by ericl** [comment]
Oh nice

File: doc/source/raysgd/raysgd_pytorch.rst [hunk 1]
```diff
@@ -11,12 +8,27 @@ Under the hood, ``PytorchTrainer`` will create *replicas* of your model (control
...
+- `PyTorch training example <https://github.com/ray-project/ray/blob/master/python/ray/util/sgd/pytorch/examples/train_example.py>`__:
+   Simple example of using Ray's PyTorchTrainer.
+
+- `CIFAR10 example <https://github.com/ray-project/ray/blob/master/python/ray/util/sgd/pytorch/examples/cifar_pytorch_example.py>`__:
+   Training a ResNet18 model on CIFAR10. It uses a custom training
+   function, a custom validation function, and custom initialization code for each worker.
+
+- `DCGAN example <https://github.com/ray-project/ray/blob/master/python/ray/util/sgd/pytorch/examples/dcgan.py>`__:
+   Training a Deep Convolutional GAN on MNIST. It constructs
+   two models and two optimizers and uses a custom training and validation function.
```
Comment (richardliaw) [inline doc/source/raysgd/raysgd_pytorch.rst]: Can you revert this particular diff? It's been added to the bottom of the page.

File: doc/source/index.rst [hunk 2]
```diff
@@ -272,31 +271,26 @@ Getting Involved
...
 
 .. toctree::
    :maxdepth: -1
-   :caption: RaySGD
+   :caption: Ray SGD
 
    raysgd/raysgd.rst
    raysgd/raysgd_pytorch.rst
    raysgd/raysgd_tensorflow.rst
-   raysgd/raysgd_ref.rst
```
Comment (richardliaw) [inline doc/source/index.rst]: Can we also keep the sgd package ref? It currently doesn't show up anywhere else.

**Comment by richardliaw** [comment]
lgtm with 2 comments

**Comment by simon-mo** [comment]
ray/util/serve -> ray/serve?

**Comment by ericl** [comment]
Looks like the tests are failing due to https://github.com/ray-project/ray/pull/7146

**Pull request merged**